src/acorn/src/expression.js

```diff
@@ -214,7 +214,12 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
 }
 
 pp.parseSubscripts = function(base, start, noCalls) {
-  if (this.eat(tt.dot)) {
+  if (this.eat(tt.doubleColon)) {
```

Comment:
In the following case:

```js
new X::y;
```

We want `(new X)::y` rather than `new (X::y)`.  Agree?

Similarly, for:

```js
x :: y :: z 
```

We want `(x :: y) :: z` rather than `x :: (y :: z)`

So I think we should only eat `::` if `noCalls` is **false**.